firebase-perf/src/main/java/com/google/firebase/perf/injection/components/FirebasePerformanceComponent.java

```diff
@@ -18,13 +18,15 @@
 import com.google.firebase.perf.FirebasePerformance;
 import com.google.firebase.perf.injection.modules.FirebasePerformanceModule;
 import dagger.Component;
+import javax.inject.Singleton;
 
 /**
  * Dagger component to create FirebasePerformanceComponent Objects.
  *
  * @hide
  */
 @Component(modules = {FirebasePerformanceModule.class})
+@Singleton
 public interface FirebasePerformanceComponent {
   @NonNull
   FirebasePerformance getFirebasePerformance();
```

Comment:
I am still new to this but shouldn't the Singleton be placed with `getFirebasePerformance()` object instead?

Reply:
Depends. In this case, both does not make a difference much. We are making the Firebase Performance Component to be a singleton as well instead of just making the instance singleton.